libs/angular/utils/src/lib/pipes/highlight/highlight.pipe.ts

```diff
@@ -1,25 +1,124 @@
 import { Pipe, PipeTransform } from '@angular/core';
+import { normalizeString } from '@studiohyperdrive/utils';
+
+import { HighlightConfig } from '../../types';
 
 @Pipe({
 	name: 'highlight',
-	standalone: true,
 })
 export class HighlightPipe implements PipeTransform {
 	/**
-	 * Highlights the provided substring of a test with the chosen dom element
+	 * Highlights the provided substring of a text with a chosen dom element
 	 *
 	 * @param value - The full text with the part we wish to highlight
 	 * @param highlight - The part of the text we wish to highlight
-	 * @param domElement - The dom-element we wish to use to highlight, by default this is a `mark` element
+	 * @param config - The configuration to determine if we want to normalize the values, to be case-sensitive, which tag and/or class to use for the highlight
+	 * @param config.normalized - Default = true
+	 * @param config.caseInsensitive - Default = true
+	 * @param config.splitTextToHighlight - Default = false
+	 * @param config.someOrEveryMatch - Default = 'every'
+	 * @param config.tag - Default = 'mark'
+	 * @param config.highlightClass - Default = 'mark-highlight'
 	 */
-	public transform(value: string, highlight: string, domElement = 'mark'): string {
-		// Iben: Early exit if there's no value or the value is not a string
-		if (!value || typeof value !== 'string') {
+	public transform(
+		value: string | null,
+		highlight: string | null,
+		config?: HighlightConfig | null
+	): string {
+		// Femke: Setup configuration or defaults
+		const normalized = config?.normalized ?? true;
+		const caseInsensitive = config?.caseInsensitive ?? true;
+		const splitTextToHighlight = config?.splitTextToHighlight ?? false;
+		const someOrEveryMatch = config?.someOrEveryMatch || 'every';
+		const tag = config?.tag || 'mark';
+		const highlightClass = config?.highlightClass ?? 'mark-highlight';
```

Comment:
We usually start our classes with `ngx-`, so that would become `ngx-mark-highlight`.